### PR TITLE
Fix for java8144+ sni changes.

### PR DESCRIPTION
**Pull request by nodefactory-bk**

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @nodefactory-bk,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by raboof** [comment]
OK TO TEST

File: akka-stream/src/main/scala/akka/stream/scaladsl/TLS.scala [hunk 1]
```diff
@@ -76,11 +76,14 @@ object TLS {
       sslConfig.getOrElse(AkkaSSLConfig(system))
 
     val createSSLEngine = { system: ActorSystem ⇒
+      val config = theSslConfig(system)
+
       val engine = hostInfo match {
-        case Some((hostname, port)) ⇒ sslContext.createSSLEngine(hostname, port)
-        case None                   ⇒ sslContext.createSSLEngine()
+        case Some((hostname, port)) if !config.config.loose.disableSNI ⇒
+          sslContext.createSSLEngine(hostname, port)
```
Comment (raboof) [inline akka-stream/src/main/scala/akka/stream/scaladsl/TLS.scala]: You mentioned you had a test case that demonstrated SNI got disabled but hostname verification was still functional - could you turn that into a unit/integrationtest?

Comment (nodefactory-bk) [reply akka-stream/src/main/scala/akka/stream/scaladsl/TLS.scala]: I could provide the test, the problem is it relies on a weirdly configured local server that I won't be able to keep publicly accessible.

And I don't think I could make a standalone test case that simulates the SNI issues without this =/

**Comment by ktoso** [comment]
Is it a java server or something we could replicate?

-- 
Cheers,
Konrad 'ktoso <http://kto.so>' Malawski
Akka <http://akka.io/> @ Lightbend <http://lightbend.com/>

On October 31, 2017 at 17:07:20, nodefactory-bk (notifications@github.com)
wrote:

*@nodefactory-bk* commented on this pull request.
------------------------------

In akka-stream/src/main/scala/akka/stream/scaladsl/TLS.scala
<https://github.com/akka/akka/pull/23878#discussion_r147917217>:

>        val engine = hostInfo match {
-        case Some((hostname, port)) ⇒
sslContext.createSSLEngine(hostname, port)
-        case None                   ⇒ sslContext.createSSLEngine()
+        case Some((hostname, port)) if !config.config.loose.disableSNI ⇒
+          sslContext.createSSLEngine(hostname, port)

I could provide the test, the problem is it relies on a weirdly configured
local server that I won't be able to keep publicly accessible.

And I don't think I could make a standalone test case that simulates the
SNI issues without this =/

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/akka/akka/pull/23878#discussion_r147917217>, or mute
the thread

**Comment by nodefactory-bk** [comment]
No, this is an apache2 server.
configured with a default virtualhost using another name so it will give the sni warning.
And also configured with a domain name that doesnt match the certificate used.
I could provide an apache config reproducing the problem with a self signed cert if that was of any help.

I can also give you the url for the server as it is right now. I can probably promise it will be up at least a week or so if this helps.

**Comment by raboof** [comment]
PLS BUILD

**Review by ktoso — APPROVED** [review]
LGTM, indeed that’s how we need to do the dance... Making a test would be possible I think, perhaps starting a server that simply has a cert to some specific cert, but we hit it on localhost? That should technically trigger the SNI check right?

**Review by patriknw — APPROVED** [review]
LGTM, I guess

**Comment by patriknw** [comment]
PLS BUILD

**Comment by ktoso** [comment]
Merged, seems okey, LGTMed and was waiting since ages

**Pull request merged**